test_runner/batch_others/test_proxy.py

```diff
@@ -22,6 +29,157 @@ def test_password_hack(static_proxy):
+        line = raw_line.decode("utf-8").strip()
+        if line.startswith("http"):
+            # poll: which code block is better? (both perform same function)
+
+            # using urlparse, 3 lines
+            url_parts = urlparse(line)
+            psql_session_id = url_parts.path[1:]
+            link_auth_uri = line[:-len(url_parts.path)]
+
+            # manual parsing, 3 lines
+            line_parts = line.split("/")
+            psql_session_id = line_parts[-1]
+            link_auth_uri = '/'.join(line_parts[:-1])
+
+            assert link_auth_uri == local_link_proxy.link_auth_uri, \
+                f"Line='{line}' should contain a http auth link of form '{local_link_proxy.link_auth_uri}/<psql_session_id>'."
+            break
+        log.debug("line %d does not contain expected result: %s", attempt, line)
+
+    assert attempt <= max_num_lines_of_welcome_message, "exhausted given attempts, did not get the result"
+    assert psql_session_id is not None, "psql_session_id not found from output of proc.stderr.readline()"
+    log.debug(f"proc.stderr.readline() #{attempt} has the result: {psql_session_id=}")
+    log.info(f"proc.stderr.readline() #{attempt} has the result: {psql_session_id=}")
+
+    return psql_session_id
+
+
+def create_and_send_db_inf(local_vanilla_pg, psql_session_id, mgmt_port):
+    pg_user = "proxy"
+    pg_password = "password"
+
+    local_vanilla_pg.start()
+    query = "create user " + pg_user + " with login superuser password '" + pg_password + "'"
+    local_vanilla_pg.safe_psql(query)
+
+    port = local_vanilla_pg.default_options['port']
+    host = local_vanilla_pg.default_options['host']
+    dbname = local_vanilla_pg.default_options['dbname']
+
+    db_info_dict = {
+        "session_id": psql_session_id,
+        "result": {
+            "Success": {
+                "host": host,
+                "port": port,
+                "dbname": dbname,
+                "user": pg_user,
+                "password": pg_password
+            }
+        }
+    }
+    db_info_str = json.dumps(db_info_dict)
+
+    # this code works, for now it's commented bc we are trying ro replace it with a psycopg2.connection
+    #
+    # mgmt_port_str = str(mgmt_port)  # hardcoded from proxy's backend I think
+    #
+    # cmd_line_args__to__mgmt = [
+    #     "psql",
+    #     "-h",
+    #     "127.0.0.1",  # localhost
+    #     "-p",
+    #     mgmt_port_str,  # mgmt port
+    #     '-c',
+    #     db_info_str
+    # ]
+    #
+    # log.info(f"Sending to proxy the user and db info: {cmd_line_args__to__mgmt}")
+    # p = subprocess.Popen(cmd_line_args__to__mgmt, stdout=subprocess.PIPE)
+    #
+    # out, err = p.communicate()
+    # log.info(f"output of sending info: out={out}; err={err}")
+    #
+    # assert "ok" in str(out)
+
+    # this code fails with E: psycopg2.DatabaseError: expected value at line 1 column 1s
+    with closing(psycopg2.connect(port=mgmt_port, host="127.0.0.1")) as conn:
```

Comment:
It's failing here with:
<img width="1193" alt="Screen Shot 2022-08-20 at 12 29 36 PM" src="https://user-images.githubusercontent.com/30240910/185932382-f1ff0626-0370-4eaa-91b8-e22d23d53de6.png">
<img width="911" alt="Screen Shot 2022-08-20 at 12 29 52 PM" src="https://user-images.githubusercontent.com/30240910/185932389-72462ba9-c410-4a3b-8e67-76c52c8fa990.png">